File: tests/bench/hardspeed_concat.ts
function buildStr(n: number): string {
    let s = "";
    for (let i = 0; i < n; i++) {
        s = s + "a";
    }
    return s;
}

for (const n of [500, 2000, 5000]) {
    const start = performance.now();
    const s = buildStr(n);
    const elapsed = (performance.now() - start) / 1000;
    console.log(`n=${n}: done (len=${s.length}) [${elapsed.toFixed(4)}s]`);
}

File: tests/bench/hardspeed_concat.py
import time

def build_str(n: int) -> str:
    s = ""
    for i in range(n):
        s = s + "a"
    return s

for n in [500, 2000, 5000]:
    start = time.time()
    s = build_str(n)
    elapsed = time.time() - start
    print(f"n={n}: done (len={len(s)}) [{elapsed:.4f}s]")
